### PR TITLE
you refill the modsuit plasma core with all your plasma at once

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

the plasma core now calculates how much plasma it needs, then uses it, rather than you having to manually put the plasma in 1 by 1

## Why It's Good For The Game

nicer to use

## Changelog

:cl:
qol: you refill the modsuit plasma core with all your plasma at once
/:cl:

### CODE REVIEW

File: code/modules/mod/mod_core.dm [hunk 1]
```diff
@@ -330,11 +330,12 @@
...
 	return NONE
 
-/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/attacking_item, mob/user)
-	if(!istype(attacking_item, /obj/item/stack/ore/plasma))
+/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/stack/plasma, mob/user)
+	if(!istype(plasma))
 		return FALSE
-	if(!attacking_item.use(1))
+	var/uses_needed = min(plasma.amount, round((max_charge_amount() - charge_amount()) / charge_given))
+	if(!plasma.use(uses_needed))
```
Comment (JohnFulpWillard) [inline code/modules/mod/mod_core.dm:337]: If you don't have enough plasma to fill it all in one go, is it just impossible to refill it?

File: code/modules/mod/mod_core.dm [hunk 2]
```diff
@@ -330,11 +330,12 @@
 		return COMPONENT_NO_AFTERATTACK
 	return NONE
 
-/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/attacking_item, mob/user)
-	if(!istype(attacking_item, /obj/item/stack/ore/plasma))
+/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/stack/plasma, mob/user)
```
Comment (JohnFulpWillard) [inline code/modules/mod/mod_core.dm:333]:
```suggestion
/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/stack/ore/plasma/plasma, mob/user)
```

Comment (Fikou) [inline code/modules/mod/mod_core.dm:333]:
```suggestion
/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/stack/ore/plasma/plasma, mob/user)
```

**Comment by Pinkufeck** [comment]
thanks nerd <3

**Pull request merged**